Robot2018/src/org/usfirst/frc/team199/Robot2018/commands/Autonomous.java

```diff
@@ -1,76 +1,45 @@
 package org.usfirst.frc.team199.Robot2018.commands;
 
+import java.util.Map;
+
 import edu.wpi.first.wpilibj.DriverStation;
 import edu.wpi.first.wpilibj.command.CommandGroup;
+import edu.wpi.first.wpilibj.command.WaitCommand;
 
 /**
  * Initially run during Auto. Responsible for getting input from SmartDashboard
  * and the FMS, and then calling RunAuto with the specified script.  
  */
 public class Autonomous extends CommandGroup implements AutonomousInterface {
 
-    public Autonomous() {
-    		// TODO
-    	
-        // Add Commands here:
-        // e.g. addSequential(new Command1());
-        //      addSequential(new Command2());
-        // these will run in order.
-
-        // To run multiple commands at the same time,
-        // use addParallel()
-        // e.g. addParallel(new Command1());
-        //      addSequential(new Command2());
-        // Command1 and Command2 will run in parallel.
-
-        // A command group will require all of the subsystems that each member
-        // would require.
-        // e.g. if Command1 requires chassis, and Command2 requires arm,
-        // a CommandGroup containing them would require both the chassis and the
-        // arm.
+    public Autonomous(String startPos, Map<String, String> strategies, double delay, String fmsInput) {
```

Comment:
Needs:
1. javadoc comments, especially describing the parameters and their expected formats.
2. At least basic input validation.
3. Unit tests covering the generation of the scriptName.

Reply:
Input validation not needed since input is given through a drop down menu and not a String input in SmartDashboard, so checking if menu items match will suffice. 

I feel like the code is trivial enough and the input restricted enough that tests shouldn't be a priority for this command.

Reply:
Short answer: If the input is from a dropdown menu, it should be trivial to validate that it is one of the values from that menu, and if the code is trivial enough, the tests should be pretty trivial to write. Regardless, it still needs the javadoc comments I mentioned.
Longer answer regarding validation: In the heat of battle someone may decide to temporarily hardcode creation of an Autonomous object. If that person accidentally passes a null ref for startPos, strategies, or fmsInput, won't the current code not only fail but also render the robot inoperable for the entire match? If you didn't know the implementations of any other classes, how confident are you that passing "Do Nothing" for the value of a strategy will in fact not do anything and not cause an error? I had to go look at the current implementation of another class. Do you see why?

Longer answer regarding unit tests: One reason to write unit tests even when doing trivial string comparisons like this is to catch typos in strings by forcing yourself to enter the strings twice -- once in the code and once in the test. If either has a typo, the tests will probably fail. If you had mistyped "Scale" as "Switch" (e.g. by copy/pasting and then getting distracted before you had a chance to change it), you or others could waste a lot of time trying to figure out why the robot is trying to do the scale even when it shouldn't. If you think that such a typo is unlikely, consider that you actually made almost exactly this typo in this file. It just happens to not be in place that causes the code to fail. Can you find it? It should be fixed.